src/libraries/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/SharedPerformanceCounter.cs

```diff
@@ -313,13 +313,13 @@ private int CalculateMemoryNoBoundsCheck(int oldOffset, int totalSize, out int a
                 newCounterEntryPointer->CounterNameHashCode = GetWstrHashCode(counterName);
                 SetValue(newCounterEntryPointer, 0);
                 newCounterEntryPointer->CounterNameOffset = (int)(nextPtr - _baseAddress);
-                SafeMarshalCopy(counterName, checked((IntPtr)nextPtr));
+                SafeMarshalCopy(counterName, (nint)nextPtr);
 
                 nextPtr += (counterName.Length + 1) * 2;
                 previousCounterEntryPointer->NextCounterOffset = (int)((long)newCounterEntryPointer - _baseAddress);
             }
 
-            Debug.Assert(nextPtr - _baseAddress == freeMemoryOffset + totalSize + alignmentAdjustment, "We should have used all of the space we requested at this point");
+            Debug.Assert((long)nextPtr - _baseAddress == freeMemoryOffset + totalSize + alignmentAdjustment, "We should have used all of the space we requested at this point");
```

Comment:
Could you please change `_baseAddress` type to `byte*` and delete these casts to `long` throughout the file.